apps/angular/permissions/src/app/routes.ts

```diff
@@ -1,14 +1,51 @@
-export const APP_ROUTES = [
+import { inject } from '@angular/core';
+import { Route } from '@angular/router';
+import { HasPermissionGuard } from './has-permission.guard';
+
+export const APP_ROUTES: Route[] = [
   {
     path: '',
     loadComponent: () =>
       import('./login.component').then((m) => m.LoginComponent),
   },
+  {
+    path: 'no-user',
+    loadComponent: () => import('./dashboard/no-user.component'),
+  },
   {
     path: 'enter',
-    loadComponent: () =>
-      import('./dashboard/admin.component').then(
-        (m) => m.AdminDashboardComponent,
-      ),
+    canMatch: [() => inject(HasPermissionGuard).canMatch()],
+    data: {
+      isAdmin: true,
+    },
```

Comment:
this should be a function param, data is not type safe

Reply:
done